devtools/config-doc-maven-plugin/src/main/java/io/quarkus/maven/config/doc/generator/GenerationReport.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.maven.config.doc.generator;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.quarkus.annotation.processor.documentation.config.model.SourceElementType;
+
+public class GenerationReport {
+
+    private Map<String, List<GenerationError>> errors = new LinkedHashMap<>();
+
+    void addError(GenerationError error) {
+        this.errors.computeIfAbsent(error.sourceType(), k -> new ArrayList<>()).add(error);
+    }
+
+    public Map<String, List<GenerationError>> getErrors() {
+        return errors;
+    }
+
+    public record ConfigPropertyGenerationError(String sourceType, String sourceElement, SourceElementType sourceElementType,
```

Comment:
Using the `Error` suffix in classes could indicate a `java.lang.Error` subclass. How about something like `ConfigPropertyGenerationViolation`?

Reply:
I adjusted it even if I think `Error` is sufficiently generic to be that connoted.